server/api.go

```diff
@@ -103,21 +104,32 @@ type UserContext struct {
 // HTTPHandlerFuncWithUserContext is http.HandleFunc but with a UserContext attached
 type HTTPHandlerFuncWithUserContext func(c *UserContext, w http.ResponseWriter, r *http.Request)
 
-func (p *Plugin) attachUserContext(handler HTTPHandlerFuncWithUserContext) http.HandlerFunc {
+func (p *Plugin) attachUserContext(handler HTTPHandlerFuncWithUserContext, requireConnected bool) http.HandlerFunc {
 	return func(w http.ResponseWriter, r *http.Request) {
 		context, cancel := p.createContext(w, r)
 		defer cancel()
 
-		info, apiErr := p.getGitlabUserInfoByMattermostID(context.UserID)
-		if apiErr != nil {
-			p.writeAPIError(w, apiErr)
+		mutex, err := cluster.NewMutex(p.API, context.UserID+"-refresh-token")
```

Comment:
I am confused, is the idea to obtain a mutex for each request to the API, to avoid racing to refresh the token?

Reply:
Yes, to make it compatible with the GitLab OAuth provider using the "refresh token rotation" strategy.